object_store/object_storage_creator.py

```diff
@@ -0,0 +1,27 @@
+from object_store.aws_s3 import AwsS3
+from object_store.azure import Azure
+from object_store.min_io import MinIO
+from object_store.object_storage_type import ObjectStorageType
+
+
+class ObjectStorageCreator:
+    def __init__(self, object_storage_type):
+        self.object_storage_type = object_storage_type
+
+    def create_storage(self, *args, **kwargs):
+        if self.object_storage_type is ObjectStorageType.MINIO:
+            ip = kwargs.get('ip', None)
+            access_key = kwargs.get('access_key', None)
+            secret_key = kwargs.get('secret_key', None)
+            minio = MinIO()
+            minio.create_client(ip=ip, access_key=access_key, secret_key=secret_key)
+            return minio
+        elif self.object_storage_type is ObjectStorageType.AWS_S3:
+            aws_s3 = AwsS3()
+            aws_s3.create_client()
+            return aws_s3
+        else:
```

Comment:
Would it be wise to check 
`  elif self.object_storage_type is ObjectStorageType.AZURE:` 
? 
And throw something if it is not?

Reply:
Yes I agree. It is important to check if correct enum name was provided for all cases. This is done by enum itself as it raises AttributeEror if trying to call tthe enum name that doesn't exist. Since we use this enum in a seprate class it is wise to add checks there also and throw ValueError if wrong object storage type was provided.